deploy-config/front-end-nginx-config/nginx.conf

```diff
@@ -11,28 +11,77 @@ http {
   default_type application/octet-stream;
   include mime.types;
   sendfile on;
-
   tcp_nopush on;
   keepalive_timeout 30;
-  port_in_redirect off; # Ensure that redirects don't include the internal container PORT - 8080
+  port_in_redirect off;
+
+  # Map host to API host
+  map $host $api_host {
+    default                 https://api.fecfile.fec.gov;
+    dev.fecfile.fec.gov     https://dev-api.fecfile.fec.gov;
+    stage.fecfile.fec.gov   https://stage-api.fecfile.fec.gov;
+    test.fecfile.fec.gov    https://test-api.fecfile.fec.gov;
+  }
 
   server {
     listen {{port}};
     root fecfile-web;
     index index.html index.htm Default.htm;
+
+    set $nonce $request_id;
+
+    # CSP headers
     location = /index.html {
       add_header Cache-Control no-cache;
       add_header Access-Control-Allow-Origin null;
-      # Ensure any updates here are reflected in the local serve in angular.json
-      add_header Content-Security-Policy "default-src 'self'; script-src 'self' 'nonce-$request_id'; style-src 'self' 'nonce-$request_id'; connect-src 'self' https://dev-api.fecfile.fec.gov https://stage-api.fecfile.fec.gov https://test-api.fecfile.fec.gov https://api.fecfile.fec.gov; frame-ancestors 'none'; object-src 'none'";
+      add_header Report-To '{"group":"csp-endpoint","max_age":10886400,"endpoints":[{"url":"/csp-report"}]}';
```

Comment:
It seems like using the Report-To header is deprecated.  https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Report-To

Same with report-uri in the CSP. https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Security-Policy/report-uri

Let's use `report-to` in the CSP https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Headers/Content-Security-Policy/report-to